src/java/picard/util/JavascriptFilter.java

```diff
@@ -0,0 +1,215 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package picard.util;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+
+import javax.script.Bindings;
+import javax.script.Compilable;
+import javax.script.CompiledScript;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+import javax.script.SimpleBindings;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.filter.SamRecordFilter;
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.CollectionUtil;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFFilterHeaderLine;
+import htsjdk.variant.vcf.VCFHeader;
+import picard.vcf.filter.VariantFilter;
+
+/**
+ * Javascript filter with HEADER type containing TYPE records. contains two
+ * static method to get a SAM Read filter or a VariantFilter
+ *
+ */
+public abstract class JavascriptFilter<HEADER, TYPE> {
+	public static final String HEADER_KEY = "header";
+	/** compiled user script */
+	private CompiledScript script = null;
+
+	/** javascript engine */
+	private ScriptEngine engine = null;
+
+	/** javascript bindings */
+	protected Bindings bindings;
+
+	/** put header in the bindings */
+	protected JavascriptFilter(final HEADER h) {
+		this.bindings = new SimpleBindings();
+		this.bindings.put(HEADER_KEY, h);
+	}
+
+	/** returns key used for header binding */
+	public String getHeaderKey() {
+		return HEADER_KEY;
+	}
+
+	/** returns key used for record binding */
+	public abstract String getRecordKey();
+
+	/** Evaluates this predicate on the given argument. */
+	protected boolean accept(final TYPE record) {
+		try {
+			this.bindings.put(getRecordKey(), record);
+			Object result = this.script.eval(bindings);
+			if (result == null)
+				return false;
+			;
+			if (result instanceof Boolean) {
+				if (Boolean.FALSE.equals(result))
```

Comment:
can you replace with return Boolean.TRUE.equals(result), with no additional if? seems clearer